lib/gitlab/regex.rb

```diff
@@ -44,13 +44,20 @@ def project_path_regex_message
 
 
     def file_name_regex
-      @file_name_regex ||= /\A[a-zA-Z0-9_\-\.]*\z/.freeze
+      @file_name_regex ||= /\A[a-zA-Z0-9_\-\.\/]*\z/.freeze
```

Comment:
This is used to validate the snippet file name as well, which shouldn't contain slashes. Please create a new `file_path_regex` and use it here and also to validate directory names in the CreateDirService.